### PR TITLE
Bug 1161633 - Create bare bones app that represents theoretical lows for app startup perf r=Eli

### PR DESCRIPTION
**Pull request by lambdabaa**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2644934e7701bea36953b489d8689b898d7aff49

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a36a7aa3dff7bd31ab98cbee5ae2b8dc066dee76

File: dev_apps/test-startup-limit/js/app.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+var worker;
+
+function render() {
+  var element = document.getElementsByClassName('count')[0];
+  element.innerHTML = count;
+}
+
+function main() {
+  // Since our entire UI is in index.html, we can emit Chrome Visible.
+  perf.mark('moz-chrome-dom-loaded');
```
Comment (eliperelman) [inline dev_apps/test-startup-limit/js/app.js]: Change `moz-chrome-dom-loaded` to `navigationLoaded`.

File: dev_apps/test-startup-limit/js/app.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    count -= 1;
+    render();
+    worker.postMessage({ method: 'setCount', params: [count] });
+  });
+
+  render();
+
+  // Now that people can tick the counter up and down,
+  // we can mark the Chrome Interactive.
+  perf.mark('moz-chrome-interactive');
```
Comment (eliperelman) [inline dev_apps/test-startup-limit/js/app.js]: Change to `chromeInteractive`.

File: dev_apps/test-startup-limit/js/app.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  console.log('Staring worker...');
+  worker = new Worker('/js/worker.js');
+  worker.postMessage({ method: 'getCount' });
+  worker.onmessage = event => {
+    // We got the current counter value from the worker.
+    count = event.data;
+    initialized = true;
+    render();
+    // Now the app is 100% ready to go.
+    perf.mark('moz-app-visually-complete');
```
Comment (eliperelman) [inline dev_apps/test-startup-limit/js/app.js]: Change to `visuallyLoaded`.

File: dev_apps/test-startup-limit/js/app.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    perf.mark('moz-content-interactive');
```
Comment (eliperelman) [inline dev_apps/test-startup-limit/js/app.js]: Change to `contentInteractive`.

File: dev_apps/test-startup-limit/js/app.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    perf.mark('moz-app-loaded');
```
Comment (eliperelman) [inline dev_apps/test-startup-limit/js/app.js]: Change to `fullyLoaded`.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0c3ddff66e077ebe4361bc4d3fd8ce2a7611f027

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=77b1a2d2d07adff63344d67ed6dc79fc55927132

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=42c51300a2ff054793cf13cd14f5fc5a818753d3

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fd63b62e53d66537b4f27a7b97d4e6aa2af07d60

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fa03c672ae649bd819ab90d3f795bbb6fc71dbe4

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=049bf9ce8cd7ab3ee66ed917593f07ee547e12b7

**Pull request merged**